src/main/java/net/dries007/tfc/config/animals/MammalConfig.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Licensed under the EUPL, Version 1.2.
+ * You may obtain a copy of the Licence at:
+ * https://joinup.ec.europa.eu/collection/eupl/eupl-text-eupl-12
+ */
+
+package net.dries007.tfc.config.animals;
+
+import java.util.function.Function;
+
+import net.minecraftforge.common.ForgeConfigSpec;
+
+public record MammalConfig(AnimalConfig inner, ForgeConfigSpec.IntValue gestationDays, ForgeConfigSpec.IntValue childCount)
+{
```

Comment:
You could add one constructor to this,
```java
public MamalConfig(AnimalConfig animal, ...) {
    this(animal.familiarityCap, animal.adulthoodDays, ... );
}
```

which then removes the need for having these nested `.inner` accesses, but still only keeps the config declarations in one place.